### ISSUE TITLE
Ubuntu 19.04 Disco Dingo needs cnf (http://archive.ubuntu.com/ubuntu/dists/disco/main/cnf/)

### ISSUE THREAD

**Issue by Fake4d** [apt-mirror/apt-mirror#118]
In the new disco dingo distribution there is a new feature called cnf in the apt. The config-file is /etc/apt/apt.conf.d/50command-not-found

```
## This file is provided by command-not-found(1) to download
## Commands metadata files.

Acquire::IndexTargets {
    # The deb822 metadata files
    deb::CNF  {
        MetaKey "$(COMPONENT)/cnf/Commands-$(NATIVE_ARCHITECTURE)";
        ShortDescription "Commands-$(NATIVE_ARCHITECTURE)";
        Description "$(RELEASE)/$(COMPONENT) $(NATIVE_ARCHITECTURE) c-n-f Metadata";
    };
};

# Refresh AppStream cache when APT's cache is updated (i.e. apt update)
APT::Update::Post-Invoke-Success {
    "if /usr/bin/test -w /var/lib/command-not-found/ -a -e /usr/lib/cnf-update-db; then /usr/lib/cnf-update-db > /dev/null; fi";
};
```

It needs the sources which are now in http://archive.ubuntu.com/ubuntu/dists/disco/main/cnf/ and http://archive.ubuntu.com/ubuntu/dists/disco-updates/main/cnf/ http://archive.ubuntu.com/ubuntu/dists/disco-security/main/cnf/ and so on

It looks like the dep11 problem with 16.04 for which there was an update in apt-mirror. 

Is it possible to fix this or to workaround this? It is not possible to mirror these files with apt-mirror for me..

**Comment by josbraden** [comment]
Fixed and tested, opened PR #120

Also fixed in master on my fork!

**Comment by nntantidakis** [comment]
Still not working... Please fix it

**Comment by olaulau** [comment]
the project seems dead, you should try a fork ...

**Comment by wedge-bf** [comment]
apt-mirror in 20.04 focal/daily is missing c-n-f Metadata too

**Comment by picardflo** [comment]
Hello,

An other exemple
Ubuntu 20.04 LTS (Dev. branch) > Client side

**Server side :**
```
root@serv1:/home/servadmin# lsb_release -a
No LSB modules are available.
Distributor ID:	Raspbian
Description:	Raspbian GNU/Linux 10 (buster)
Release:	10
Codename:	buster
root@serv1:/home/servadmin#
root@serv1:/home/servadmin# apt install apt-mirror
Lecture des listes de paquets... Fait
Construction de l'arbre des dépendances       
Lecture des informations d'état... Fait
apt-mirror est déjà la version la plus récente (0.5.4-1).
0 mis à jour, 0 nouvellement installés, 0 à enlever et 0 non mis à jour.
```

**Client side :** 

```
E: Impossible de récupérer http://mirror.home.lan/mirror/fr.archive.ubuntu.com/ubuntu/dists/focal/main/cnf/Commands-amd64  404  Not Found [IP : 192.168.11.5 80]
E: Impossible de récupérer http://mirror.home.lan/mirror/fr.archive.ubuntu.com/ubuntu/dists/focal-updates/main/cnf/Commands-amd64  404  Not Found [IP : 192.168.11.5 80]
E: Impossible de récupérer http://mirror.home.lan/mirror/fr.archive.ubuntu.com/ubuntu/dists/focal-security/main/cnf/Commands-amd64  404  Not Found [IP : 192.168.11.5 80]
E: Le téléchargement de quelques fichiers d'index a échoué, ils ont été ignorés, ou les anciens ont été utilisés à la place.

```

**Comment by fe80** [comment]
Hello, you can use debmirror https://salsa.debian.org/debian/debmirror/-/tree/master if you want

**Comment by Stifler6996** [comment]
I have created a pull request to fix this.
I tested it and it worked for me.

If you want to try it, you can use the perl script in my github
https://github.com/Stifler6996/apt-mirror
It is a straight perl script, so you could just make a backup of your existing one - `/usr/bin/apt-mirror` and then copy this. Then adjust permissions of the new file.

or look in the pull requests here https://github.com/apt-mirror/apt-mirror/pull/136

Hopefully the pull-request goes through soon. But it will probably not make it all the way back to Ubuntu repo's for a long time.

**Comment by slashdd** [comment]
A mentionned in https://launchpad.net/bugs/1825755 ... I'm looking for volunteer to give a test package a try before I consider an SRU into Ubuntu.

For now the test package is offered for Bionic, but let me know (in launchpad) what release you want the test package and I'll make it available.

**Comment by slashdd** [comment]
should be now available from Bionic to Groovy (active devel release)
Feel free to to give it a try and report in launchpad please.

**Comment by slashdd** [comment]
As an fyi, the test package I made available contains josbraden's fork patchset.

**Comment by slashdd** [comment]
@josbraden can I suggest you take ownership of apt-mirror upstream project instead of updating your own fork ? Seems like upstream is seeking for a new maintainer. Just asking if it interest you.

**Comment by josbraden** [comment]
@slashdd I'll send them an email. Thanks!

**Comment by EricSP2** [comment]
> 
> 
> I have created a pull request to fix this.
> I tested it and it worked for me.
> 
> If you want to try it, you can use the perl script in my github
> https://github.com/Stifler6996/apt-mirror
> It is a straight perl script, so you could just make a backup of your existing one - `/usr/bin/apt-mirror` and then copy this. Then adjust permissions of the new file.
> 
> or look in the pull requests here #136
> 
> Hopefully the pull-request goes through soon. But it will probably not make it all the way back to Ubuntu repo's for a long time.

Great work on the fix, I swapped my apt-mirror with you code and now the cnf directory is finally sync'd and everything works 100%

Thanks!

**Comment by Rameshradhakrishnan** [comment]
> > I have created a pull request to fix this.
> > I tested it and it worked for me.
> > If you want to try it, you can use the perl script in my github
> > https://github.com/Stifler6996/apt-mirror
> > It is a straight perl script, so you could just make a backup of your existing one - `/usr/bin/apt-mirror` and then copy this. Then adjust permissions of the new file.
> > or look in the pull requests here #136
> > Hopefully the pull-request goes through soon. But it will probably not make it all the way back to Ubuntu repo's for a long time.
> 
> Great work on the fix, I swapped my apt-mirror with you code and now the cnf directory is finally sync'd and everything works 100%
> 
> Thanks!

Yes it works. good work. Thanks

**Comment by jsuelwald** [comment]
This is still a problem with a fresh Ubuntu Server 2.04-Installation.
How long does it usually take for this to be merged into the ubuntu-repositories?

**Comment by Fake4d** [comment]
> This is still a problem with a fresh Ubuntu Server 20.04-Installation.
> How long does it usually take for this to be merged into the ubuntu-repositories?

https://github.com/apt-mirror/apt-mirror/issues/139#issuecomment-698317180
Until it is working you can use a forked version - if there wont be any activity again I want to contact the MOTU people!

**Comment by kiloforce** [comment]
> I have created a pull request to fix this.
> I tested it and it worked for me.
> If you want to try it, you can use the perl script in my github
> https://github.com/Stifler6996/apt-mirror

Worked great @Stifler6996 thanks!  Was frustrated that I couldn't get my Ubuntu 20 Focal mirror to work, but you solved it.

**Comment by chutzimir** [comment]
This was fixed with #156. Can be closed.

**Comment by SupaHiro** [comment]
Thanks @chutzimir! @Fake4d I'll be closing this issue and adding it to the changelog for 0.6.0.
